TransactionListener.py

```diff
@@ -41,7 +41,23 @@ def validate_transaction(transaction):
 
 # function to verify and signatures and hashes for transactions and block data
 def validate_block(block):
-    pass
+    transactions_list = []
+    for transaction in block.transactions.values():
+        transactions_list.append(transaction)
```

Comment:
One thing which we might have to check is if the transaction order is preserved in a Python dict object, as it was in the order it occurred during the mining of the block. If there are issues, we can resolve it by keeping the `block.transactions` as a list of transaction.